File: mobile/ios/Accord/Models/Channel.swift
// Channel.swift — Text or voice channel within a Node

import Foundation

enum ChannelType: String, Codable {
    case text
    case voice
}

struct Channel: Identifiable, Codable, Hashable {
    let id: String
    var name: String
    var type: ChannelType
    var category: String?   // Grouping label
    var nodeId: String
    var topic: String?
    var isEncrypted: Bool = true
}

File: mobile/android/app/src/main/java/com/accord/data/model/Channel.kt
package com.accord.data.model

enum class ChannelType { TEXT, VOICE, DM }

data class Channel(
    val id: String,
    val name: String,
    val type: ChannelType,
    val nodeId: String? = null,
    val categoryId: String? = null,
    val position: Int = 0,
    val topic: String = "",
)
